dev/test/pool.ts

```diff
@@ -156,9 +177,134 @@ describe('Client pool', () => {
       () => operationPromises[1].promise
     );
     expect(clientPool.size).to.equal(1);
+    assertOpCount(clientPool, 2, 0);
+
+    operationPromises[0].resolve();
+    operationPromises[1].resolve();
+  });
+
+  it('does not re-use rest instance after beginning the transition to grpc', async () => {
+    const clientPool = new ClientPool<{}>(10, 1, () => {
+      return {};
+    });
+
+    const operationPromises = deferredPromises(3);
+
+    void clientPool.run(
+      REQUEST_TAG,
+      USE_REST,
+      () => operationPromises[0].promise
+    );
+    void clientPool.run(
+      REQUEST_TAG,
+      USE_GRPC,
+      () => operationPromises[1].promise
+    );
+    void clientPool.run(
+      REQUEST_TAG,
+      USE_REST,
+      () => operationPromises[2].promise
+    );
+
+    expect(clientPool.size).to.equal(2);
+    assertOpCount(clientPool, 2, 1);
 
     operationPromises[0].resolve();
     operationPromises[1].resolve();
+    operationPromises[2].resolve();
+  });
+
+  it('does not re-use rest instance after beginning the transition to grpc - rest operation resolved', async () => {
+    const clientPool = new ClientPool<{}>(10, 1, () => {
+      return {};
+    });
+
+    const operationPromises = deferredPromises(3);
+
+    const restOperation = clientPool.run(
+      REQUEST_TAG,
+      USE_REST,
+      () => operationPromises[0].promise
+    );
+    void clientPool.run(
+      REQUEST_TAG,
+      USE_GRPC,
+      () => operationPromises[1].promise
+    );
+
+    // resolve rest operation
+    operationPromises[0].resolve();
+    await restOperation;
+    expect(clientPool.opCount).to.equal(1);
+
+    // Run new rest operation
+    void clientPool.run(
+      REQUEST_TAG,
+      USE_REST,
+      () => operationPromises[2].promise
+    );
+
+    // Assert client pool status
+    expect(clientPool.size).to.equal(1);
+    assertOpCount(clientPool, 2, 0);
+
+    operationPromises[1].resolve();
+    operationPromises[2].resolve();
+  });
+
+  it('does not re-use rest instance after beginning the transition to grpc - grpc client full', async () => {
```

Comment:
Can we also add another test in which REST client is full and results in a second REST client, followed by a grpc operation?

e.g. 15 rest ops + 5 grpc ops (with limit of 10) --> client 1: 10 rest ops, client 2: 5 rest ops, client 3: 5 grpc ops

Reply:
Thanks for the suggestion. I added this.

Reply:
👍